app/controllers/concerns/foreman/controller/parameters/subnet.rb

```diff
@@ -34,4 +34,19 @@ def subnet_params_filter
   def subnet_params
     self.class.subnet_params_filter.filter_params(params, parameter_filter_context)
   end
+
+  def network_reorder(base)
```

Comment:
It feels like this should actually be a Model concern.

Reply:
I tried this approach, I failed to find a way to hook into scoped search, scoped search does not provide a way to implement casting, especially not for just some adapters. Then I thought let's do it in model, but I'd need to extend Subnet::ActiveRecord_Relation, not Subnet class. And then I gave up.

Now I realize that we can actually define it in Subnet model and it works just fine. So here you are, thanks for pushing me :-)

```ruby
Subnet.all.network_reorder
```

```ruby
Subnet.all.network_reorder 'network DESC'
```